### ISSUE TITLE
[ML/Dynamics] Extend Patchbay watches for tensors, trajectories, uncertainty, state, loss, and learned dynamics

### ISSUE THREAD

**Issue by dancxjo** [dancxjo/conduit#2144]
Parent: #2132
Depends on: #2133 #2134 #2135 #2138 #2140 #2141
Related: real-time Patchbay debugger/watch work

## Outcome

Extend Patchbay's runtime observation/watch model so learned and dynamical Forms can be debugged as **live evolving systems**, not merely as nodes that occasionally emit opaque blobs.

The motivating experience is to watch Tongues run and visibly inspect:

- PCM/audio flow;
- EMA/articulatory trajectories;
- latent `z(t)` dimensions;
- recurrent state generations;
- uncertainty/posterior envelopes;
- sampled alternatives;
- loss/objective components;
- parameter/checkpoint transitions;
- continuous-dynamics integration;
- phase/coupling analysis outputs.

This should build on the existing debug-observation boundary rather than create a privileged ML control plane.

## Background

The existing Patchbay direction already calls for a real-time debugger in which data visibly flows through the diagram using animation, coloring, text, and explicit watches.

The ML/dynamics substrate makes this more demanding because values may be:

- high-rank tensors too large to dump;
- time-indexed streaming blocks;
- probabilistic rather than deterministic;
- recurrent state whose generation matters;
- model/checkpoint/resource references;
- continuous trajectories where scrolling/time alignment is more useful than one textual value.

Patchbay therefore needs **bounded projections appropriate to value semantics**, not generic `Debug` strings.

## Watch architecture

A watch should remain an explicitly admitted observation of runtime truth.

Suggested separation:

```text
runtime value/sign/state
      |
      v
bounded debug observation
      |
      v
watch projection request
      |
      v
Patchbay renderer-specific visualization
```

The watch/projection layer must never mutate or bypass the mandatory Sign path.

## Tensor watch

For #2134 tensor values, support bounded summaries such as:

- dtype;
- shape;
- axis roles/units;
- byte size;
- min/max/mean/std where explicitly computed by an admitted bounded observation operation;
- bounded sample/slice;
- small matrix/vector rendering where under limits;
- resource/checkpoint identity where payload is referenced.

Do not automatically materialize gigabyte tensors to render a watch.

## Signal/trajectory watch

For #2135 signals:

- scrolling time-series plot;
- current clock/start/rate/continuity;
- selected semantic channels;
- alignment of several watched signals by a known ClockRelation;
- clear visual gaps for discontinuities/missing samples;
- pause/freeze/inspect history under finite retained-history limits;
- current-value readout at selected time.

Use the generic watch path for PCM, F0, articulator coordinates, latent variables, loss trajectories, and sensor streams.

## Probabilistic watch

For #2138 outputs:

- distinguish observed vs inferred vs sampled values visually/textually;
- posterior mean plus uncertainty band where available;
- weighted sample alternatives;
- sample count;
- seed/stochastic profile;
- approximation/truncation status;
- never label a posterior sample as “actual articulation.”

## State watch

For #2133:

- current state generation/step;
- current value identity;
- pending candidate next state;
- commit transition;
- reset/cancel/refusal;
- bounded state summaries.

This should make optimizer/model-state loops legible without turning state into ordinary combinational cycle arrows.

## Training watch

For #2140:

- active training/evaluation state;
- batch/example/split identity summary;
- objective components and total loss;
- step count/work accounting;
- checkpoint creation events;
- resource pressure/refusal;
- evaluation metrics;
- cancellation/failure.

Avoid a dashboard that implies Patchbay itself owns training state. It is observing/projecting the Body's actual execution.

## Dynamics watch

For #2141:

- initial/final state summary;
- requested interval;
- output trajectory;
- solver work count;
- tolerance/error summary;
- truncation/work-bound failure;
- optional vector-field/phase portrait visualization when semantically supported and bounded.

Do not equate solver internal microsteps with semantic samples unless explicitly exposed by the result contract.

## Time alignment UX

Tongues debugging particularly needs several clocks on screen at once.

Patchbay should be able to show:

```text
audio clock A
EMA clock B
latent/model clock C
```

and, when a known relation exists, align derived/rendered views while preserving which source clock each value actually belongs to.

A missing ClockRelation should be visible as “not aligned,” not silently guessed.

## Dataflow animation

Use the existing runtime debugger goal to show value movement through Cords, but avoid making high-rate streaming produce unusable per-sample animation.

Need finite aggregation/rate controls such as:

- block-level activity;
- throughput/rate summaries;
- most recent value identity;
- selected watch channels;
- bounded retained history.

Backpressure/refusal/loss/discontinuity should have distinct projections.

## Performance/boundedness

Observation must not materially destabilize the system being observed.

Define limits for:

- watches per Play/subject;
- retained samples/history bytes;
- tensor slice bytes;
- update rate;
- plot point count/downsampling;
- probabilistic sample count rendered;
- remote observation bandwidth.

Dropping presentation updates must not mutate semantic execution; debug observation should report its own pressure/loss where material.

## Native/browser consistency

Patchbay model/projection semantics should remain renderer-neutral where possible. Browser and native renderers may use different plotting libraries, but watch identity and bounded observation contracts should not fork.

## Non-goals

- no unrestricted memory inspection;
- no arbitrary GPU tensor-pointer access;
- no unbounded training telemetry stream;
- no Patchbay-owned optimizer/model mutation in this issue;
- no replacement of Sign/provenance with debug telemetry;
- no assumption all tensor dimensions are visually meaningful;
- no automatic alignment of clocks without explicit relation.

## Acceptance criteria

- [ ] Watch model supports tensor summaries without unbounded materialization.
- [ ] Generic signal/trajectory plots work for at least audio-derived, articulatory, latent, and metric streams.
- [ ] Probabilistic watches visually distinguish posterior/sampled information from observation.
- [ ] State generation/commit transitions are visible.
- [ ] Training loss/objective/checkpoint events are visible.
- [ ] Dynamics integration trajectory/work/error can be watched.
- [ ] Multiple signals can align only through explicit clock-relation evidence.
- [ ] Observation history/update rates are bounded and pressure-aware.
- [ ] Runtime semantics remain authoritative; Patchbay is a projection/debugger.
- [ ] Browser and native paths consume the same watch/projection model where practical.
- [ ] A Tongues-shaped demo shows live audio, articulator trajectory, latent trajectory, posterior uncertainty, and loss simultaneously while data visibly moves through the Form.

## Product test

The experience should answer, at a glance:

> **What information is moving through this learned system right now, what does the model believe, what state changed, and where did the computation become surprising?**

**Comment by dancxjo** [comment]
Starting from exact `main` commit `73685a5fc2fe7546aed5ee86b561af534acb8ce7`.

Outcome: extend the existing kernel-observation-backed `DebuggerWatchSet` with a finite renderer-neutral learned/dynamics projection vocabulary, render it through the real Patchbay HTML snapshot/runtime, and prove a Tongues-shaped live view plus refusal/boundedness cases.

Owned files: focused modules/tests under `apps/patchbay/model`, the existing Patchbay HTML snapshot/demo/assets, and focused real-Patchbay browser proof.

Non-goals / stop line: no second Patchbay/runtime/dashboard, no training or model mutation, no unrestricted tensor materialization, no guessed clock alignment, and no claim that presentation telemetry replaces Signs or semantic execution. Acceptance requires focused Rust contract tests, the strict Patchbay browser proof, exact-head required CI, and exact-main rerun before closure.

**Comment by dancxjo** [comment]
Accepted on exact `main` commit `0f21edf5a876733364f2f4e274a7fdeb270e6510` via merged PR #2198.

Exact-head evidence:
- required `check` run https://github.com/dancxjo/conduit/actions/runs/33706786211 passed, including all six applicable workspace/lint shards; unrelated browser-host, browser-tools, ESP32, and ConduitOS product/proof jobs skipped;
- focused real-Patchbay browser run https://github.com/dancxjo/conduit/actions/runs/33706786209 passed 5/5 with one Chromium worker and zero retries.

Fresh exact-main evidence at `0f21edf5`:
- `cargo +stable test -p patchbay-model --locked learned_watch` — 3 passed;
- `cargo +stable test -p patchbay-html --locked learned_demo --jobs 1` — learned fixture snapshot round-trip passed;
- `cargo +stable build --locked -p patchbay-html` — product built;
- `npx playwright test --config proof/browser/patchbay-debugger.config.mjs proof/browser/patchbay-debugger-watch.spec.mjs --project chromium --workers 1 --retries 0` — 5/5 passed.

The accepted product path is the existing authoritative Patchbay: exact kernel observations feed `DebuggerWatchSet`, learned projections are sequence-bound and finite, newer runtime truth invalidates older projections, and the HTML renderer consumes that same model. The Tongues-shaped proof simultaneously renders audio-derived, articulatory, latent, and metric trajectories; posterior uncertainty and alternatives; state commit; objective/loss/checkpoint/pressure; dynamics work/error; explicit clock relation versus not-aligned state; missing samples; and the existing animated Cord. Patchbay does not own or mutate execution.

No Pages deployment is claimed by this issue: this is a loopback-server Patchbay debugger slice. Browser-hostable learned artifacts remain downstream work, beginning with #2145/#2146.
